ha_lineairdb.cc

```diff
@@ -814,19 +814,33 @@ std::string ha_lineairdb::get_current_key() {
   return std::string(current_key_.ptr(), current_key_.length());
 }
 
+std::array<char, BYTE_BIT_NUMBER> ha_lineairdb::interpret_buf_flag(uchar flag) {
+  std::array<char, BYTE_BIT_NUMBER> byte_nullflag = {};
+  for(int i = 0; i < 8; i++) {
+    byte_nullflag[i] = (flag >> i) & 1;
+  }
+  return byte_nullflag;
+}
+
 /**
  * @brief Format and set the requested row into `write_buffer_`.
  * A MySQL Row is serialized to comma-separated double-quoted values,
  * as the followings:
  * {"alice","bob","carol","3"}
  *   col1    col2  col3   col4
  */
-void ha_lineairdb::set_write_buffer() {
+void ha_lineairdb::set_write_buffer(uchar *buf) {
   write_buffer_.length(0);
 
   char attribute_buffer[1024];
   String attribute(attribute_buffer, sizeof(attribute_buffer), &my_charset_bin);
 
+  std::array<char, BYTE_BIT_NUMBER> byte_nullflag = interpret_buf_flag(buf[0]);
+  /* index to store the null flag */
+  int buf_nullbyte_index = 0;
+  /* index to store the bit wihtin 1 byte flag */
+  int nullable_field_index = 0;
```

Comment:
Use `size_t` to indicate to collaborators that this is a monotonically increasing index for array.
This also helps the performance. See https://in-neuro.hatenablog.com/entry/2020/02/07/200259